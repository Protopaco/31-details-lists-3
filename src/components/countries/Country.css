@import url('../../context/variables.css');


.image{
    height: 125px;
}
.figure{
    padding: 25px 25px;
    border: .5px solid var(--secondary);
    border-radius: 15px;
    background: var(--primary);
    width: 250px;
    height: 150px;
    display: flex;
    flex-direction: column;
    align-items: center;
    color: var(--secondary);
    font-style: bold;
}
.caption{
    margin-top: 10px;
}
