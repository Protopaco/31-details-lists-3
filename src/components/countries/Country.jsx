import React from 'react';
import PropTypes from 'prop-types';
import styles from './Country.css'

const Country = ({ name, flagURL }) => (
    <figure className={styles.figure}>
        <img src={flagURL} alt={name} className={styles.image} />
        <figcaption className={styles.caption}>{name}</figcaption>
    </figure>
)

Country.propTypes = {
    flagURL: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}

export default Country
