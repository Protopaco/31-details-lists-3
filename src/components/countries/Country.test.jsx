import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Country from './Country';

describe('Country component', () => {
    afterEach(() => cleanup());
    it('renders Country', () => {
        const { asFragment } = render(<Country name={'f'} flagURL={'f'} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
