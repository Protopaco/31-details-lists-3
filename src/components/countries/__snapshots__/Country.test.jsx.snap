// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component renders Country 1`] = `
<DocumentFragment>
  <figure
    class="figure"
  >
    <img
      alt="f"
      class="image"
      src="f"
    />
    <figcaption
      class="caption"
    >
      f
    </figcaption>
  </figure>
</DocumentFragment>
`;
